benchmark/cpp_mergeRandom_readRandom.sh

```diff
@@ -0,0 +1,257 @@
+# benchmarks  -- see each benchmark
+
+num=$((1 * 1000 * 1000))
+reads=$num
+key_size=2
+value_size=$((4 * 1024))
+
+# Number of concurrent threads to run for the benchmark.
+threads=2
+
+
+############################################
+####   Statistics                       ####
+############################################
+# Report database statistics (true/false)
+statistics=1
+
+# Print histogram of operation timings (true/false)
+histogram=1
+
+# a temp varibale that estimates the total size of an entry
+entry_size=$(($key_size + $value_size))
+
+# Stats are reported every N operations when this is greater than zero.
+# When 0 the interval grows over time. The formula below comes out of an
+# experiment where we had an 10M interval with 16B keys and 8B values.
+# interval*entry_size=10M*(16+8), hence interval = 240M/entry_size
+stats_interval=$((240000000 / $entry_size))
+
+# Reports additional stats per interval when this is greater than 0.
+stats_per_interval=1
+
+
+############################################
+####   DB                               ####
+############################################
+db=$1
+
+if [ ! -d "$db" ]; then
+  mkdir -p $db
+fi
+
+# use_existing_db  -- see each benchmark
+
+############################################
+####   Parallelism                      ####
+############################################
+# maximum number of concurrent background compactions
+max_background_compactions=1
+
+# maximum number of concurrent flush operations. 
+max_background_flushes=1
+
+
+############################################
+####   General                          ####
+############################################
+# Default bits_per_key is 10, which yields ~1% false positive rate. Larger
+# values will reduce false positive rate, but increase memory usage and
+# space amplification.
+# Bloom filter bits per key. Negative means use default settings
+bloom_bits=10
+
+# RocksDB keeps all file descriptors in a table cache. If number of file
+# descriptors exceeds max_open_files, some files are evicted from table cache
+# and their file descriptors closed. Every read must go through the table
+# cache to lookup the file needed. Set max_open_files to -1 to always keep
+# all files open, which avoids expensive table cache calls.
+max_open_files=100000
+
+# RocksDB packs user data in blocks. When reading a key-value pair from a
+# table file, an entire block is loaded into memory. 
+block_size=$(($entry_size * 16))
+
+############################################
+####   Flushing (memtables)             ####
+############################################
+# memtable size in bytes
+write_buffer_size=$((1024 * 1024))
+
+# If the active memtable fills up and the total number of memtables is
+# larger than max_write_buffer_number we stall further writes
+max_write_buffer_number=6
+
+# minimum number of memtables to be merged before flushing to storage
+min_write_buffer_number_to_merge=2
```

Comment:
I think max_background_flushes should be larger than one so that all 5 of the non-active write buffers can be flushed concurrently (set it to between 3 and 5).